Refines Principal Software Engineer role definition (#61)
OctopusDeploy/People: Software-Engineering/L5-Principal-Software-Engineer.md

## 🦉 Domain expertise

- I contribute to core technologies used by a variety of teams.
- I help teams make critical product architecture and implementation decisions.
- I maintain a vision of the overall technical direction within my impact radius.
- I contribute to the engineering standards and principles of the company.
- I contribute to the Octopus technical strategy.
- I manage technical risks, and architect to reduce technical risk.
- I collaborate with other Principal Software Engineers to ensure we are making cohesive and complimentary decisions that align with the Octopus technical strategy.

<details>
<summary>Examples</summary>

- I successfully completed complex tasks spanning multiple domains and teams with high impact. 
- I helped a team reduce complexity and risk through sound architectural thinking.

> - I developed a plan to evolve the architecture of one of the core components of Octopus within my impact radius.

Reviewer: Key part of this is `within my impact radius`  👍  .

One problem I found previously was being expected to develop and lead improvements to core architecture but being linked to a team which didn't necessitate many "core architectural" changes.  

I.e. The difference between a platform team vs product team are going to have much different impacts in this space.

- I identified a problem that was timely, important, and impactful, and worked closely with product, engineering, and executive stakeholders to articulate and refine a solution and delivery plan.
- I worked with teams to manage multiple conflicting priorities and helped sequence work resulting in the best collective outcome.

> - I collaborated with another principal engineer to ensure that a feature I was helping a team with would support a feature I knew another team was planning.

Reviewer 2: 🤔 I'm not sure we can dictate the scope of projects. Would it be enough for us to connect the relevant teams so they can make the best decision?

Author: I don't think we can dictate it, but I think we can influence it from time to time, and that influence can be important and timely.

> - I ensured the feature I was helping a team plan made decisions that supported a feature I knew another team was planning by collaborating with that team's aligned principal engineer.

Reviewer 3: I found this sentence difficult to parse; here's a reword to consider:
```suggestion
- I collaborated with another principal engineer to ensure that a feature I was helping a team with would support a feature I knew another team was planning.
```

<summary>Examples</summary>

- I was consistently in demand for design and code review.
- I am actively mentoring multiple Octopus engineers.
- Engineers at Octopus seek out to work on my teams to learn from me.

> - After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager to uplift their skills in the identified growth area.

Reviewer 2: This one seems to be very prescriptive and might not scale.  E.g. I usually work with L4 running the project/team to ensure the uplift happens. L4s should already have a great communication channel established with their manager. 

A more generic version:

` After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager/lead to uplift their skills in the identified growth area.`

> - After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager to uplift their skills in the identified growth area.

Author: I'll update this with "lead" as well 👍

> - After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager to uplift their skills in the identified growth area.

Author: ```suggestion
- After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager or team leader to uplift their skills in the identified growth area.
```

- I helped a team make a course correction based on new information or ideas.
- I provided appropriate technical freedom, and framing, that allowed the Lead Software Engineers on my teams to thrive independently.

</details>

## 🧭 Culture and Leadership

> - I model our cultural safety by thinking in the open, asking imperfect questions, being wrong, and accepting corrections.

Reviewer 3: Terms like "model our cultural safety" and "imperfect questions" felt a bit abstract/open to interpretation. Here's a reword to consider, which puts the intended _outcome_ upfront:

```suggestion
- I create an environment where others feel safe, by modeling vulnerability, thinking in the open, being wrong, and accepting corrections.
```

> - I create an environment where others feel safe, by modeling vulnerability, thinking in the open, being wrong, and accepting corrections.

Reviewer 4: I like this one.

Reviewer 5: +1

- I generate energy within the teams I work with, motivating them to achieve great outcomes.
- I facilitate effective and timely decision making, especially when the decision affects multiple teams.

> - I build solutions that improve the productivity and quality of many people/teams.

Reviewer 3: ```suggestion
- I build solutions that improve the productivity and/or quality of work of many people/teams.
```

- I collaborated with other senior leaders to ensure I was aware of all major initiatives at Octopus and could account for them in my own initiatives.
- I identified a significant problem/opportunity and created a lasting best-fit solution that aligned with our strategy.
- Instead of simply patching or extending an existing solution that was not fit for purpose anymore, my contribution opened up a whole new area of strategic possibility.

> - I participate in our code review and interview processes, and provide feedback as to how they can be improved regularly.

Reviewer 3: [nitpick] Prefer past-tense for concrete `Examples`:
```suggestion
- I regularly participated in our code review and interview processes and provided feedback on how they could be improved.
```
